src/agentscope/memory/reme/_reme_personal_memory.py

```diff
@@ -0,0 +1,398 @@
+
+            # Extract metadata about stored memories if available
+            metadata = result.get("metadata", {})
+            memory_list = metadata.get("memory_list", [])
+
+            if memory_list:
+                summary_text = (
+                    f"Successfully recorded {len(memory_list)} memory/memories "
+                    f"to personal memory."
+                )
+            else:
+                summary_text = "Memory recording completed."
+
+            return ToolResponse(
+                content=[
+                    TextBlock(
+                        type="text",
+                        text=summary_text,
+                    ),
+                ],
+                metadata={"result": result},
+            )
+
+        except Exception as e:
+            logger.exception("Error recording memory: %s", str(e))
+            return ToolResponse(
+                content=[
+                    TextBlock(
+                        type="text",
+                        text=f"Error recording memory: {str(e)}",
+                    ),
+                ],
+            )
+
+    async def retrieve_from_memory(
+        self,
+        keywords: list[str],
+        **kwargs: Any,
+    ) -> ToolResponse:
+        """Search and retrieve relevant information from long-term memory.
+
+        IMPORTANT: You should call this function BEFORE answering questions
+        about the user's preferences, past information, or personal details.
+        This ensures you provide accurate information based on stored memories
+        rather than guessing.
+
+        Use this when:
+        - User asks "what do I like?", "what are my preferences?", "what do you know about me?"
+        - User asks about their past behaviors, habits, or stated preferences
+        - User refers to information they shared in previous conversations
+        - You need to personalize responses based on user's history
+
+        Args:
+            keywords (`list[str]`):
+                Keywords to search for in memory. Be specific and use multiple
+                keywords for better results. Examples: ["travel preferences",
+                "Hangzhou"], ["work habits", "morning routine"], ["food preferences",
+                "tea"].
+            **kwargs (`Any`):
+                Additional keyword arguments for the retrieval operation.
+
+        Returns:
+            `ToolResponse`:
+                Retrieved memories matching the keywords. If no memories found,
+                you'll receive a message indicating that.
+        """
+        logger.info(
+            "[ReMePersonalMemory] Entering retrieve_from_memory - "
+            "keywords: %s, kwargs: %s",
+            keywords,
+            kwargs,
+        )
+
+        if not self._app_started:
+            return ToolResponse(
+                content=[
+                    TextBlock(
+                        type="text",
+                        text="Error: ReMeApp context not started. "
```

Comment:
This error is developer-oriented, so it should be raised as an exception rather than presented to the agent?